src/lib.rs

```diff
@@ -270,20 +273,23 @@ enum IpfsEvent {
 /// Configured Ipfs instace or value which can be only initialized.
 pub struct UninitializedIpfs<Types: IpfsTypes> {
     repo: Option<Repo<Types>>,
+    span: Span,
     keys: Keypair,
     options: IpfsOptions<Types>,
     moved_on_init: Option<Receiver<RepoEvent>>,
 }
 
 impl<Types: IpfsTypes> UninitializedIpfs<Types> {
-    /// Configures a new UninitializedIpfs with from the given options.
-    pub async fn new(options: IpfsOptions<Types>) -> Self {
+    /// Configures a new UninitializedIpfs with from the given options and optionally a span.
+    pub async fn new(options: IpfsOptions<Types>, span: Option<Span>) -> Self {
         let repo_options = RepoOptions::<Types>::from(&options);
         let (repo, repo_events) = create_repo(repo_options);
         let keys = options.keypair.clone();
+        let span = span.unwrap_or_else(|| tracing::trace_span!("ipfs"));
```

Comment:
I wonder if this should be `span.unwrap_or_default()` instead? Though I guess using a span always would be a net benefit; all paths should use the span and so by being by-default correlatable it would make a lot of sense.

Reply:
However if this is anything other than the empty span, as discussed, we need to make sure all tasks spawned by libp2p via the `Executor` or whatever the almost `futures` `Spawner` trait was called are instrumented.

Reply:
`Span` doesn't impl `Default` though

Reply:
> we need to make sure all tasks spawned by libp2p via the Executor or whatever the almost futures Spawner trait was called are instrumented

confirmed :+1: